xdsl/ir.py

```diff
@@ -640,7 +639,7 @@ class Block(IRNode):
 
     declared_at: 'Span' | None = None
 
-    _args: FrozenList[BlockArgument] = field(default_factory=FrozenList,
+    _args: tuple[BlockArgument, ...] = field(default_factory=lambda: (),
```

Comment:
Why the lambda here and `tuple` above?

Reply:
My vague recolllection is that the type system complained but I can check again
- [ ] Check whether `tuple` woudl work here

Reply:
ok turns out it doesn't work above either, it complains that the type of `_args` is partially unknown when I pass the `tuple` function directly